src/cmap/commands.ts

```diff
@@ -730,10 +731,19 @@ const COMPRESSION_DETAILS_SIZE = 9; // originalOpcode + uncompressedSize, compre
  * An OP_COMPRESSED request wraps either an OP_QUERY or OP_MSG message.
  */
 export class OpCompressedRequest {
+  private command: WriteProtocolMessageType;
+  private options: { zLibCompressionLevel: number; agreedCompressor: CompressorName };
+
   constructor(
-    private command: WriteProtocolMessageType,
-    private options: { zlibCompressionLevel: number; agreedCompressor: CompressorName }
-  ) {}
+    command: WriteProtocolMessageType,
+    options: { zlibCompressionLevel: number; agreedCompressor: CompressorName }
+  ) {
+    this.command = command;
+    this.options = {
+      zLibCompressionLevel: options.zlibCompressionLevel,
```

Comment:
```suggestion
      zlibCompressionLevel: options.zlibCompressionLevel,
```

Reply:
should there be some sort of test that catches this?